### PR TITLE
bau: Migrate WorldpayPaymentProviderTest to Junit5

### PR DESCRIPTION
**Pull request by oswaldquek**

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java [hunk 1]
```diff
@@ -356,12 +363,12 @@ public void shouldNotIncludeElementsWhenWorldpay3dsFlexDdcResultIsNotPresent() t
         assertThat(xPath.evaluate("/paymentService/submit/order/shopper/browser/userAgentHeader", document),
                 not(emptyString()));
     }
-
+    
     @Test
-    public void shouldInclude3DS2FlexElementsWhenWorldpay3dsFlexDdcResultIsPresent() throws Exception {
+    void should_include_3DS2_flex_elements_when_worldpay_3ds_flex_ddc_resultI_is_present() throws Exception {
```
Comment (nimalank7) [inline src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java]: Should be should_include_3DS2_flex_elements_when_worldpay_3ds_flex_ddc_result_is_present()

Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java]: Ugh, now I have to wait for a whole build again :(

Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java]: Fixed

File: src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java [hunk 2]
```diff
@@ -513,7 +521,7 @@ public void shouldErrorIfWorldpayReturns401() throws Exception {
     }
```
Comment (nimalank7) [inline src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java:522]: On Lines 518-520 I think we could use the JUnit 5 `assertThrows` and remove the try-catch block with something like this?
```suggestion
var thrown = assertThat(GatewayError.class, () -> providerWithRealGatewayClient.authorise(getCardAuthorisationRequest());
assertThat(thrown.getMessage(), is(...));
    }
```

Comment (nimalank7) [reply src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java:522]: Similarly with any other tests which assert on exceptions.

**Comment by nimalank7** [comment]
Mostly good! Left a comment about using JUnit 5 assertThrows.

File: src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java (hunk 2 above)
Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gateway/worldpay/WorldpayPaymentProviderTest.java:522]: Fixed

**Review by nimalank7 — APPROVED** [review]
LGTM!

**Pull request merged**